test/swar/BasicOperations.cpp

```diff
@@ -4,9 +4,192 @@
+            1           0       x0      1       01111...
+            1           0       x01     2       00111...
+            0           x01
+3           3           0       x0      1       01111...
+            1           x0      x1      1       01111...
+            1           x0      x12     2       00111...
+            0           x012    x23
+4           4           0       x0      1       01111...
+            2           0       x0      1       01111...
+            2           0       x01     2       00111...
+            1           0       x01
+            1           0       x0123   4       00001...
+            0           x0123   x01234567
+5           5           0       x0      1       01111...
+            2           x0      x1
+            2           x0      x12     2       00111...
+            1
+            1                   x1234   4       00001...
+            0           x01234
+6           6           0       x0      1       01111...
+            3
+            3                   x01     2       00111...
+            1           x01     x23     
+            1                   x2345   4       00001...
+            0           x012345 x6789
+7           7           0       x0      1       01......
+            3           x0      x1
+            3                   x12     2       001.....
+            1           x012    x34
+            1                   x3456   4       00001...
+            0           x0-6    x789A
+25 = 16 + 8 + 1
+25          25          0       x0      1       01111...
+            12          x0      x1
+            12                  x12     2       00111...
+            6
+            6                   x1234   4       {0}4
+            3
+            3                   x1-8    8       {0}8
+            1           x0-8    x9-16
+            1                   x9-24   16      {0}16
+            0           x0-24   x25-?
+*/
+
+template<int NB, typename B>
+struct ArithmeticResultTriplet {
+    SWAR<NB, B> result;
+    BooleanSWAR<NB, B> carry, overflow;
+};
+
+template<int NB, typename B>
+constexpr auto makeElementMaskFromLSB(SWAR<NB, B> input) {
+    auto lsb = input.value() & SWAR<NB, B>::LeastSignificantBit;
+    // copy the LSB to the MSB position
+    auto msb = lsb << (NB - 1);
+    // subtracting one we "copy" the msb to all lower positions, but the
+    // msb is cleared: 0x80 -> 0x7F
+    // subtracting zero there is no change
+    auto zeroOrMsbCopiedDown = msb - lsb;
+    // copy again the msb: 0x7F | 0x80 = 0xFF
+    return SWAR<NB, B>(zeroOrMsbCopiedDown | msb);
+}
+
+template<int NB, typename B>
+constexpr auto fullAddition(SWAR<NB, B> s1, SWAR<NB, B> s2) {
+    using S = SWAR<NB, B>;
+    constexpr auto
+        SignBit = S{S::MostSignificantBit},
+        LowerBits = SignBit - S{S::LeastSignificantBit};
+    // prevent overflow by clearing the most significant bits
+    auto
+        s1prime = LowerBits & s1,
```

Comment:
using 'prime' to not mean 'prime number' in this context is going to confuse people.